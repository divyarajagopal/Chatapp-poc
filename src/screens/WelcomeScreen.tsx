import * as React from 'react';
import { View } from 'react-native';
import UserRegistrationUI from '../components/UserRegistration/UserRegistrationUI';

export class WelcomeScreen extends React.Component {
  constructor(public props: any) {
    super(props);
  }
  public render() {
    return (
      <View style={{ flex: 1 }}>
        <UserRegistrationUI navigation={this.props.navigation} />
      </View>
    );
  }
}

export default WelcomeScreen;
