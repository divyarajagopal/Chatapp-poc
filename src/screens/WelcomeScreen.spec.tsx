import * as React from 'react';
import configureStore from 'redux-mock-store';
import { IApplicationStoreState } from '../__common__';
import { WelcomeActionTypes } from '../__common__/redux-modules/welcome';
import { shallow, configure } from 'enzyme';
import { WelcomeScreen } from './WelcomeScreen';
import toJson from 'enzyme-to-json';
import * as Adapter from 'enzyme-adapter-react-16';
import { Provider } from 'react-redux';
import * as renderer from 'react-test-renderer';

describe('WelcomeScreen Component', () => {
  configure({ adapter: new Adapter() });
  const mockStore = configureStore();
  let store: any, tree: any;
  beforeEach(() => {
    const applicationState: IApplicationStoreState = {
      welcomeStoreState: {
        data: [],
        errorMessage: undefined,
        isLoaded: false
      }
    };
    store = mockStore(applicationState);
  });

  test(`should render the component as expected`, () => {
    tree = shallow(<WelcomeScreen />);
    expect(toJson(tree)).toMatchSnapshot();
  });

  test('should render the component as expected with store context', () => {
    const wrapper = shallow(<WelcomeScreen />, { context: { store } });
    expect(wrapper.dive()).toMatchSnapshot();
  });

  test('should dispatch actions correctly on load', () => {
    tree = renderer
      .create(
        <Provider store={store}>
          <WelcomeScreen />
        </Provider>
      )
      .toJSON();

    const actions = store.getActions();
    expect(actions.length).toBe(2);
    expect(actions[0].type).toBe(WelcomeActionTypes.LOADING_WELCOME_SCREEN);
    expect(actions[1].type).toBe(WelcomeActionTypes.LOADED_WELCOME_SCREEN);
  });
});
