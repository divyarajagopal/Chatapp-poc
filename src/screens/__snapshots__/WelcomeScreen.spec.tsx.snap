// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WelcomeScreen Component should render the component as expected 1`] = `
<Component
  style={
    Object {
      "flex": 1,
    }
  }
>
  <UserRegistrationUI />
</Component>
`;

exports[`WelcomeScreen Component should render the component as expected with store context 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
